docs/adapters/custom/parallel-requests.md

```diff
@@ -42,18 +42,34 @@ class FlorpHttp < ::Faraday::Adapter
   def self.setup_parallel_manager(_options = nil)
     FlorpParallelManager.new # NB: we will need to define this
   end
-end
 
-class FlorpParallelManager
-  def add(request, method, *args, &block)
-    # Collect the requests
+  def call(env)
+    # NB: you can call `in_parallel?` here to check if the current request
+    # is part of a parallel batch. Useful if you need to collect all requests
+    # into the ParallelManager before running them.
   end
+end
 
-  def run
-    # Process the requests
+class FlorpParallelManager
+  # The execute method will be passed the same block as `in_parallel`,
+  # so you can either collect the requests or just wrap them into a wrapper,
+  # depending on how your adapter works.
+  def execute(&block)
+    run_async(&block)
   end
 end
 ```
 
-Compare to the finished example [em-synchrony](https://github.com/lostisland/faraday-em_synchrony/blob/main/lib/faraday/adapter/em_synchrony.rb)
+### A note on the old, deprecated interface
+
+Prior to the introduction of the `execute` method, the `ParallelManager` was expected to implement a `run` method
+and the execution of the block was done by the Faraday connection BEFORE calling that method.
+
+This approach made the `ParallelManager` implementation harder and keeping the state required.
```

Comment:
```suggestion
This approach made the `ParallelManager` implementation harder and keeping keeping state.
```

Reply:
@olleolleolle did you want to suggest a different thing here? `keeping keeping` doesn't really make sense so I guess this is a typo (guess: only one `keeping`?)

Reply:
yes, hehe, keep one _keeping_

Reply:
@iMacTia There, more clear, I hope.